torch/csrc/jit/runtime/profiling_record.h

```diff
@@ -27,6 +81,58 @@ struct ProfilingRecord {
   std::shared_ptr<Graph> profiled_graph_;
   std::mutex mutex_;
   size_t profiling_count_;
+  std::map<int64_t, std::map<Value*, TensorTypePtr>> profiled_types_per_frame_;
```

Comment:
I can't follow this data structure in the code. There are no comments about what it stores, and it results in a lot of "x.first" and "x.second" confusion.